lib/src/components/NxFilterInput/__tests__/NxFilterInput.test.tsx

```diff
@@ -5,160 +5,108 @@
-  it('renders a PrivateNxTextInput with the nx-filter-input class', function() {
-    const component = shallowComponent();
-    expect(component).toHaveClassName('nx-filter-input');
-    expect(component).toMatchSelector(PrivateNxTextInput);
+  const minimalProps: Props = { value: '' },
+      quickRender = rtlRender(NxFilterInput, minimalProps),
+      renderEl = rtlRenderElement(NxFilterInput, minimalProps);
+
+  it('renders an Input', function() {
+    expect(quickRender().getByRole('textbox').tagName).toBe('INPUT');
   });
 
-  it('adds custom classnames as specified', function() {
-    const component = shallowComponent({ className: 'foo' });
-    expect(component).toHaveClassName('foo');
-    expect(component).toHaveClassName('nx-filter-input');
+  it('adds additional specified classNames to the top-level element', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
+
+    expect(el).toHaveClass('foo');
+
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('passes misc props to the PrivateNxTextInput', function() {
-    const onChange = jest.fn(),
-        onKeyPress = jest.fn(),
-        component = shallowComponent({
-          value: 'foo',
-          onChange,
-          onKeyPress,
-          id: 'bar'
-        });
-
-    expect(component).toHaveProp('value', 'foo');
-    expect(component).toHaveProp('onChange', onChange);
-    expect(component).toHaveProp('onKeyPress', onKeyPress);
-    expect(component).toHaveProp('id', 'bar');
+  it('passes additional attrs to the Input', function() {
+    const input = quickRender({ id: 'foo', lang: 'en-US' }).getByRole('textbox');
+
+    expect(input).toHaveAttribute('id', 'foo');
+    expect(input).toHaveAttribute('lang', 'en-US');
   });
 
-  it('does not pass validatable, validationErrors, or type props to the PrivateNxTextInput', function() {
-    const component = shallowComponent({
+  it('does not pass validatable, validationErrors, or type props to the Input', function() {
+    const input = quickRender({
       validatable: true,
       validationErrors: 'It\'s all wrong',
       type: 'textarea'
-    } as Partial<Props>);
-
-    expect(component).not.toHaveProp('validatable');
-    expect(component).not.toHaveProp('validationErrors');
-    expect(component).not.toHaveProp('type');
-  });
+    } as Partial<Props>).getByRole('textbox');
 
-  it('passes isPristine = false to the PrivateNxTextInput', function() {
-    expect(shallowComponent()).toHaveProp('isPristine', false);
+    expect(input).not.toHaveAttribute('validatable', true);
+    expect(input).not.toHaveAttribute('validationErrors', 'It\'s all wrong');
+    expect(input).not.toHaveAttribute('type', 'textarea');
   });
 
-  it('puts the ref on the PrivateNxTextInput', function() {
+  it('sets ref on the Input', function() {
     const ref = React.createRef<HTMLDivElement>(),
+        el = renderEl({ ref } as Partial<Props>);
```

Comment:
If you change the type of `minimalProps` to `Props & RefAttributes<HTMLDivElement>` then you don't need this cast.  `RefAttributes` is an export from `react`.

Reply:
Thanks for the info. Updated here 7cd3180bedc9a6842ee950fc523e59059ac33e60.